src/resources/2021/DaylightShavings.ts

```diff
@@ -0,0 +1,66 @@
+import { myClass, toEffect, toInt } from "kolmafia";
+import { have as haveItem } from "../../lib";
+import { get } from "../../property";
+import { $effects, $item } from "../../template-string";
+
+export const helmet = $item`Daylight Shavings Helmet`;
+
+/**
+ * Returns whether the player owns an unpackaged Daylight Shavings Helmet, and it's available in either the inventory or other zones as determined by autoSatisfy settings.
+ * @returns whether we have the Daylight Shavings Helmet.
+ */
+export function have(): boolean {
+  return haveItem(helmet);
+}
+
+export const buffs = $effects`Spectacle Moustache, Toiletbrush Moustache, Barbell Moustache, Grizzly Beard, Surrealist's Moustache, Musician's Musician's Moustache, Gull-Wing Moustache, Space Warlord's Beard, Pointy Wizard Beard, Cowboy Stache, Friendly Chops`;
+
+export function hasBuff(): boolean {
+  return buffs.some((buff) => haveItem(buff));
+}
+/**
+ * Calculates and returns the cycle of buffs that the hat should cycle through.
+ * @param playerclass The class to generate a cycle for
+ * @returns An ordered array consisting of the cycle for this class. The first element of the array will be the first buff a player should expect to get in a given ascension.
+ */
+export function buffCycle(playerclass = myClass()): Effect[] {
+  if (toInt(playerclass) <= 0) return [];
+  const returnValue: Effect[] = [];
+  const id = toInt(playerclass);
+  const seed = id > 6 ? (id % 6) + 1 : id;
+  for (let i = 1; i < 12; i++) {
+    const index = (i * seed) % 11;
+    returnValue.push(buffs[index]);
+  }
+  return returnValue;
+}
+
+/**
+ * Returns the next buff we expect to get from the shaving hat.
+ * @returns The next buff we expect to get from the shaving hat.
+ */
+export function nextBuff(): Effect {
+  const currentBuff = toEffect(get("lastBeardBuff").toFixed(0));
+
+  const cycle = buffCycle();
+
+  const index = cycle.indexOf(currentBuff);
+
+  const newIndex = (1 + index) % 11;
+  return cycle[newIndex];
+}
+
+/**
```

Comment:
i think all these functions make sense but buffsUntil is a little unintuitive, though i see why it is useful after our convo on discord. i would recommend a different docstring. here is my attempt to punch it up. 
```/**
 * Returns the number of shaving hat activations we expect it'll take to get to a given buff, given our knowledge about the user's current buffCycle. Returns 1 for the next buff, 2 for the one after that, and so on. Returns 11 for the most recent buff. This is useful for both ascertaining whether or not to buy SGEEAs to cycle to your next buff and timing out how long you would need to wait without SGEEA interaction.
 * @param buff The shaving buff in question (as an $effect)
 * @returns The number of shaving hat activations necessary to get the shaving buff in @param; null if said buff is not granted by the shaving hat.
 */```